### ISSUE TITLE
Typescript crashing or so

### ISSUE THREAD

**Issue by aronmal** [microsoft/vscode#139875]
Issue Type: <b>Bug</b>

I don't know the issue, i only have a continuous loading ring on the bottem left corner with 'Initializing JS/TS language features' and i got a pop up, which said typescipt crashed more then 5 times in the last 5 minutes. I case i broke something, here is a report so you can maybe make it more idiot proof ;) Have a nice day!

VS Code version: Code 1.63.2 (899d46d82c4c95423fb7e10e68eba52050e30ba3, 2021-12-15T09:39:46.686Z)
OS version: Linux x64 5.15.5-76051505-generic
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz (12 x 3700)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|31.21GB (21.16GB free)|
|Process Argv|--unity-launch --crash-reporter-id 966b8802-50c2-4c25-a8d6-9953802a11ec|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|pop|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|pop|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (2)</summary>

Extension|Author (truncated)|Version
---|---|---
es7-react-js-snippets|dsz|3.1.1
material-icon-theme|PKi|4.11.0

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
pythontb:30283811
pythonvspyt551cf:30345471
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
vscop804cf:30404767
vscop341:30404997
vsrem710:30416614
vscexrecpromp3t2:30407763

```

</details>

**Comment by aronmal** [comment]
The pop up:

![image](https://user-images.githubusercontent.com/57275207/147654012-fd796fec-346c-4442-82d8-dc5db40ab0db.png)

**Comment by aronmal** [comment]
On my bottom left corner:

![image](https://user-images.githubusercontent.com/57275207/147654139-f387a849-9556-4e5e-acae-5b9d7604cf26.png)

**Comment by IllusionMH** [comment]
/needsMoreInfo

Please follow steps described in https://github.com/microsoft/vscode/issues/139093#issuecomment-994182490 to get error from logs.

**Comment by aronmal** [comment]
Here is the log file:

[tsserver.log](https://github.com/microsoft/vscode/files/7788879/tsserver.log)

**Comment by aronmal** [comment]
> /needsMoreInfo
> 
> Please follow steps described in [#139093 (comment)](https://github.com/microsoft/vscode/issues/139093#issuecomment-994182490) to get error from logs.

I did:

> Here is the log file:
> 
> [tsserver.log](https://github.com/microsoft/vscode/files/7788879/tsserver.log)


Do i have to do something else?

**Comment by mjbvz** [comment]
In VS Code, please use the `toggle developer tools` command to open the developer tools. Then in the console, try looking for errors related to the typescript server. The most interesting crash is likely the first one

If you have a project that triggers this issue reliably and can share it, we can also use that to investigate the problem

**Comment by aronmal** [comment]
> In VS Code, please use the `toggle developer tools` command to open the developer tools. Then in the console, try looking for errors related to the typescript server. The most interesting crash is likely the first error. Please share the text of this error message so we know what is going wrong (unfortunately the tsserver log you uploaded doesn't include the cause of the crash )
> 
> If you have a project that triggers this issue reliably and can share it, we can also use that to investigate the problem

Should i upload the project as a zip file or can i invite you to my private Gtihub repo? (It is a tool to download HLS content with nodejs, as early information)

**Comment by aronmal** [comment]
For now, I tried recreating it, but for now it still works. I have to mention, that I already continued my project, so the code is different than before, but even with the 'old' code without the new commits, it does not show an error when opening the project (unlike before, where I was able to recreate it). I am also not sure, but I worked with nodemon and saved the file multiple times while changing the code, which caused nodemon telling me that my application crashed and after I fixed the unfinished problems my application started and worked as it should during the development. Because of the warning, which I got, I tried sharing it in case it is providing information to you (But it was only opening an Github Issue to my surprise). If I should have the same warning again, I will contact you back.
